jsonrpc/server.go

```diff
@@ -300,6 +304,19 @@ func buildArguments(params, handler any, configuredParams []Parameter) ([]reflec
 			return reflect.ValueOf(nil), err
 		}
 
+		// If the argument is a struct, validate it.
+		value := handlerParam
+		// If the argument to the handler is a pointer, we need to dereference
+		// one time since we initialized handlerParam with reflect.New(t).
+		if t.Kind() == reflect.Pointer {
+			value = value.Elem()
+		}
+		if err = validate.Struct(value.Interface()); err != nil {
```

Comment:
We can just pass `handlerParam.Elem().Interface()` here, no? Without any of the checks above.

handlerParam is a pointer since it is created by `reflect.New()`. `Elem()` will dereference it and return the value of type that the handler expects it to be in. If the resulting parameter is a pointer `validate.Struct` will deference it to validate the underlying value.